src/compiler/checker.ts

```diff
@@ -4542,7 +4542,7 @@ namespace ts {
             return type;
         }
 
-        function getTypeFromStringLiteral(node: StringLiteral): Type {
+        function getTypeFromStringLiteral(node: StringLiteral | StringLiteralTypeNode): Type {

```

Comment:
I'll put a comment in checkStringLiteralExpression instead. The big difference is that getTypeFromStringLiteral caches the type on the resolvedType of the node's NodeLinks.